diff --git a/node_modules/azure-devops-ui/Components/MouseWithin/MouseWithin.js b/node_modules/azure-devops-ui/Components/MouseWithin/MouseWithin.js
index 871fd62..a4a9a60 100644
--- a/node_modules/azure-devops-ui/Components/MouseWithin/MouseWithin.js
+++ b/node_modules/azure-devops-ui/Components/MouseWithin/MouseWithin.js
@@ -26,15 +26,19 @@ var MouseWithin = /** @class */ (function (_super) {
                     _this.delayTimeout = 0;
                 }
                 if (_this.props.enterDelay) {
-                    event.persist();
-                    // persist does not preserve the currentTarget so we do that manually
-                    var currentTarget_1 = event.currentTarget;
+                    /* this patch makes a shallow clone of the event
+                     * as in this repo we use preact which does not have
+                     * a synthetic events like in react. This change in behaviour
+                     * results in an exception that currentTarget is only a getter
+                     *  */
+                    var currentEvent = {
+                        currentTarget: event.currentTarget,
+                        target: event.target,
+                        nativeEvent: event.nativeEvent,
+                    };
                     _this.enterTimeout = window.setTimeout(function () {
                         _this.enterTimeout = 0;
-                        var newCurrentTarget = event.currentTarget;
-                        event.currentTarget = currentTarget_1;
-                        _this.mouseEntered(event);
-                        event.currentTarget = newCurrentTarget;
+                        _this.mouseEntered(currentEvent);
                     }, _this.props.enterDelay);
                 }
                 else {
